packages/indexer-agent/src/network.ts

```diff
@@ -857,6 +857,72 @@ export class Network {
     }
   }
 
+  async previousAllocations(
+    subgraphDeploymentId: SubgraphDeploymentID,
+  ): Promise<Allocation[]> {
+    try {
+      const result = await this.networkSubgraph.query(
+        gql`
+          query allocations($indexer: String!, $subgraphDeploymentId: String!) {
+            indexer(id: $indexer) {
+              allocations: totalAllocations(
+                where: {
+                  indexer: $indexer
+                  status: Closed
+                  subgraphDeployment: $subgraphDeploymentId
+                }
+                first: 5
+                orderBy: closedAtEpoch
```

Comment:
I know the contracts don't allow more than one allocation in the same epoch for a single deployment, but just to be safer and bit more explicit let's order by `closedAtBlockNumber`?